### ISSUE TITLE
feature_loader.rb throwing unexpected errors on GitHub actions, trying to load plugins not referenced

### ISSUE THREAD

**Issue by thomthom** [rubocop/rubocop#12592]
I have a project where `bundle exec rubocop` work fine without any errors or warnings on both Windows and macOS locally.

But running on GitHub Actions it's throwing errors that's very unexpected. 

On `ubuntu-latest` it first emits a list of newer cops to enable, even though `NewCops` is disabled. But the main issue is that it's throwing this error: `cannot load such file -- rubocop-rubycw`

Looking at the source of `feature_loader.rb` I don't understand what/why it trying to load that. I have no reference to `rubocop-rubycw` in my project.

`ubuntu-latest`
https://github.com/SketchUp/testup-2/actions/runs/7397583032/job/20124958959#step:4:251
```
cannot load such file -- rubocop-rubycw
/home/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/rubocop-1.59.0/lib/rubocop/feature_loader.rb:46:in `rescue in rescue in load'
/home/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/rubocop-1.59.0/lib/rubocop/feature_loader.rb:39:in `rescue in load'
/home/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/rubocop-1.59.0/lib/rubocop/feature_loader.rb:32:in `load'
/home/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/rubocop-1.59.0/lib/rubocop/feature_loader.rb:21:in `load'
/home/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/rubocop-1.59.0/lib/rubocop/config_loader_resolver.rb:14:in `block (2 levels) in resolve_requires'
/home/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/rubocop-1.59.0/lib/rubocop/config_loader_resolver.rb:13:in `each'
```

---

Similarly, on `macos-latest` it's trying (and failing) to load `rubocop-minitest`.

Any ideas to why rubocop might try to load these plugins when the project doesn't refer to them at all? Having a hard time to debug this since it happens only on the CI agents, not on my local machines (Windows, macOS)

`macos-latest`
https://github.com/SketchUp/testup-2/actions/runs/7397641120/job/20125130357#step:4:7
```
cannot load such file -- rubocop-minitest
/Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.[7](https://github.com/SketchUp/testup-2/actions/runs/7397641120/job/20125130357#step:4:8).0/gems/rubocop-1.5[9](https://github.com/SketchUp/testup-2/actions/runs/7397641120/job/20125130357#step:4:10).0/lib/rubocop/feature_loader.rb:46:in `rescue in rescue in load'
/Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/rubocop-1.59.0/lib/rubocop/feature_loader.rb:39:in `rescue in load'
/Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/rubocop-1.59.0/lib/rubocop/feature_loader.rb:32:in `load'
/Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/rubocop-1.59.0/lib/rubocop/feature_loader.rb:21:in `load'
/Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/rubocop-1.59.0/lib/rubocop/config_loader_resolver.rb:14:in `block (2 levels) in resolve_requires'
/Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/rubocop-1.59.0/lib/rubocop/config_loader_resolver.rb:13:in `each'
/Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/rubocop-1.59.0/lib/rubocop/config_loader_resolver.rb:13:in `block in resolve_requires'
/Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/rubocop-1.59.0/lib/rubocop/config_loader_resolver.rb:[12](https://github.com/SketchUp/testup-2/actions/runs/7397641120/job/20125130357#step:4:13):in `tap'
/Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/rubocop-1.59.0/lib/rubocop/config_loader_resolver.rb:12:in `resolve_requires'
```

--------

## Expected behavior

I would expect RuboCop to successfully run and pass the checks.

## Actual behavior

It throws errors while booting up trying to load RuboCop plugins that are not referenced in the project.

```
Run bundle exec rubocop --debug
  bundle exec rubocop --debug
  shell: /bin/bash -e {0}
  env:
    PATH: /usr/local/opt/pipx_bin:/Users/runner/.cargo/bin:/usr/local/opt/curl/bin:/usr/local/bin:/usr/local/sbin:/Users/runner/bin:/Users/runner/.yarn/bin:/Users/runner/Library/Android/sdk/tools:/Users/runner/Library/Android/sdk/platform-tools:/Library/Frameworks/Python.framework/Versions/Current/bin:/Library/Frameworks/Mono.framework/Versions/Current/Commands:/usr/bin:/bin:/usr/sbin:/sbin:/Users/runner/.dotnet/tools:/Users/runner/.ghcup/bin
Unprocessable file /Users/runner/work/testup-[2](https://github.com/SketchUp/testup-2/actions/runs/7397830754/job/20125680317#step:5:2)/testup-2/tests/SketchUp Ruby API/TC_Sketchup_ImageRep/emptyfile: EOFError, end of file reached
cannot load such file -- rubocop-minitest
/Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/rubocop-1.59.0/lib/rubocop/feature_loader.rb:46:in `rescue in rescue in load'
/Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/rubocop-1.59.0/lib/rubocop/feature_loader.rb:[3](https://github.com/SketchUp/testup-2/actions/runs/7397830754/job/20125680317#step:5:3)9:in `rescue in load'
/Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/rubocop-1.59.0/lib/rubocop/feature_loader.rb:32:in `load'
/Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/rubocop-1.59.0/lib/rubocop/feature_loader.rb:21:in `load'
/Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/rubocop-1.59.0/lib/rubocop/config_loader_resolver.rb:1[4](https://github.com/SketchUp/testup-2/actions/runs/7397830754/job/20125680317#step:5:4):in `block (2 levels) in resolve_requires'
/Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/rubocop-1.[5](https://github.com/SketchUp/testup-2/actions/runs/7397830754/job/20125680317#step:5:5)9.0/lib/rubocop/config_loader_resolver.rb:13:in `each'
/Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/rubocop-1.59.0/lib/rubocop/config_loader_resolver.rb:13:in `block in resolve_requires'
/Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/rubocop-1.59.0/lib/rubocop/config_loader_resolver.rb:12:in `tap'
/Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/rubocop-1.59.0/lib/rubocop/config_loader_resolver.rb:12:in `resolve_requires'
/Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/rubocop-1.59.0/lib/rubocop/config_loader.rb:53:in `load_file'
/Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/rubocop-1.59.0/lib/rubocop/config_loader.rb:112:in `configuration_from_file'
/Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/rubocop-1.59.0/lib/rubocop/config_store.rb:[6](https://github.com/SketchUp/testup-2/actions/runs/7397830754/job/20125680317#step:5:7)8:in `for_dir'
/Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.[7](https://github.com/SketchUp/testup-2/actions/runs/7397830754/job/20125680317#step:5:8).0/gems/rubocop-1.59.0/lib/rubocop/config_store.rb:5[8](https://github.com/SketchUp/testup-2/actions/runs/7397830754/job/20125680317#step:5:9):in `for'
/Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/rubocop-1.5[9](https://github.com/SketchUp/testup-2/actions/runs/7397830754/job/20125680317#step:5:10).0/lib/rubocop/target_finder.rb:165:in `ruby_interpreters'
/Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/rubocop-1.59.0/lib/rubocop/target_finder.rb:157:in `ruby_executable?'
/Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/rubocop-1.59.0/lib/rubocop/target_finder.rb:173:in `ruby_file?'
/Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/rubocop-1.59.0/lib/rubocop/target_finder.rb:73:in `to_inspect?'
/Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/rubocop-1.59.0/lib/rubocop/target_finder.rb:64:in `block in target_files_in_dir'
/Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/rubocop-1.59.0/lib/rubocop/target_finder.rb:64:in `select'
/Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/rubocop-1.59.0/lib/rubocop/target_finder.rb:64:in `target_files_in_dir'
/Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/rubocop-1.59.0/lib/rubocop/target_finder.rb:32:in `find'
/Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/rubocop-1.59.0/lib/rubocop/runner.rb:[11](https://github.com/SketchUp/testup-2/actions/runs/7397830754/job/20125680317#step:5:12)1:in `find_target_files'
/Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/rubocop-1.59.0/lib/rubocop/runner.rb:68:in `run'
/Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/rubocop-1.59.0/lib/rubocop/cli/command/execute_runner.rb:26:in `block in execute_runner'
/Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/rubocop-1.59.0/lib/rubocop/cli/command/execute_runner.rb:52:in `with_redirect'
/Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/rubocop-1.59.0/lib/rubocop/cli/command/execute_runner.rb:25:in `execute_runner'
/Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/rubocop-1.59.0/lib/rubocop/cli/command/execute_runner.rb:17:in `run'
/Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/rubocop-1.59.0/lib/rubocop/cli/command.rb:11:in `run'
/Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/rubocop-1.59.0/lib/rubocop/cli/environment.rb:18:in `run'
/Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/rubocop-1.59.0/lib/rubocop/cli.rb:118:in `run_command'
/Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/rubocop-1.59.0/lib/rubocop/cli.rb:[12](https://github.com/SketchUp/testup-2/actions/runs/7397830754/job/20125680317#step:5:13)5:in `execute_runners'
/Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/rubocop-1.59.0/lib/rubocop/cli.rb:51:in `block in run'
/Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/rubocop-1.59.0/lib/rubocop/cli.rb:77:in `profile_if_needed'
/Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/rubocop-1.59.0/lib/rubocop/cli.rb:43:in `run'
/Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/rubocop-1.59.0/exe/rubocop:19:in `block in <top (required)>'
/Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/benchmark-0.3.0/lib/benchmark.rb:3[13](https://github.com/SketchUp/testup-2/actions/runs/7397830754/job/20125680317#step:5:14):in `realtime'
/Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/rubocop-1.59.0/exe/rubocop:19:in `<top (required)>'
/Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/bin/rubocop:23:in `load'
/Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/bin/rubocop:23:in `<top (required)>'
/Users/runner/hostedtoolcache/Ruby/2.7.8/x64/lib/ruby/gems/2.7.0/gems/bundler-2.4.22/lib/bundler/cli/exec.rb:58:in `load'
/Users/runner/hostedtoolcache/Ruby/2.7.8/x64/lib/ruby/gems/2.7.0/gems/bundler-2.4.22/lib/bundler/cli/exec.rb:58:in `kernel_load'
/Users/runner/hostedtoolcache/Ruby/2.7.8/x64/lib/ruby/gems/2.7.0/gems/bundler-2.4.22/lib/bundler/cli/exec.rb:23:in `run'
/Users/runner/hostedtoolcache/Ruby/2.7.8/x64/lib/ruby/gems/2.7.0/gems/bundler-2.4.22/lib/bundler/cli.rb:492:in `exec'
/Users/runner/hostedtoolcache/Ruby/2.7.8/x64/lib/ruby/gems/2.7.0/gems/bundler-2.4.22/lib/bundler/vendor/thor/lib/thor/command.rb:28:in `run'
/Users/runner/hostedtoolcache/Ruby/2.7.8/x64/lib/ruby/gems/2.7.0/gems/bundler-2.4.22/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
/Users/runner/hostedtoolcache/Ruby/2.7.8/x64/lib/ruby/gems/2.7.0/gems/bundler-2.4.22/lib/bundler/vendor/thor/lib/thor.rb:527:in `dispatch'
/Users/runner/hostedtoolcache/Ruby/2.7.8/x64/lib/ruby/gems/2.7.0/gems/bundler-2.4.22/lib/bundler/cli.rb:34:in `dispatch'
/Users/runner/hostedtoolcache/Ruby/2.7.8/x64/lib/ruby/gems/2.7.0/gems/bundler-2.4.22/lib/bundler/vendor/thor/lib/thor/base.rb:584:in `start'
/Users/runner/hostedtoolcache/Ruby/2.7.8/x64/lib/ruby/gems/2.7.0/gems/bundler-2.4.22/lib/bundler/cli.rb:28:in `start'
/Users/runner/hostedtoolcache/Ruby/2.7.8/x64/lib/ruby/gems/2.7.0/gems/bundler-2.4.22/exe/bundle:37:in `block in <top (required)>'
/Users/runner/hostedtoolcache/Ruby/2.7.8/x64/lib/ruby/gems/2.7.0/gems/bundler-2.4.22/lib/bundler/friendly_errors.rb:117:in `with_friendly_errors'
/Users/runner/hostedtoolcache/Ruby/2.7.8/x64/lib/ruby/gems/2.7.0/gems/bundler-2.4.22/exe/bundle:29:in `<top (required)>'
/Users/runner/hostedtoolcache/Ruby/2.7.8/x64/bin/bundle:23:in `load'
/Users/runner/hostedtoolcache/Ruby/2.7.8/x64/bin/bundle:23:in `<main>'
For /Users/runner/work/testup-2/testup-2: configuration from /Users/runner/work/testup-2/testup-2/.rubocop.yml
configuration from /Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/rubocop-performance-1.[15](https://github.com/SketchUp/testup-2/actions/runs/7397830754/job/20125680317#step:5:16).2/config/default.yml
configuration from /Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/rubocop-performance-1.15.2/config/default.yml
Default configuration from /Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/rubocop-1.59.0/config/default.yml
configuration from /Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/rubocop-sketchup-1.3.0/config/default.yml
configuration from /Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/rubocop-sketchup-1.3.0/config/default.yml
Use parallel by default.
For /Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/skippy-0.5.2.a/bin: configuration from /Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/2.7.0/gems/skippy-0.5.2.a/.rubocop.yml
Finished in 0.8[26](https://github.com/SketchUp/testup-2/actions/runs/7397830754/job/20125680317#step:5:27)83900000006[36](https://github.com/SketchUp/testup-2/actions/runs/7397830754/job/20125680317#step:5:37) seconds
Error: Process completed with exit code 2.
```

## Steps to reproduce the problem

I'm having a hard time reproducing locally. :/

## RuboCop version

`macos-latest`:
```
$ [bundle exec] rubocop -V
1.59.0 (using Parser 3.2.2.4, rubocop-ast 1.30.0, running on ruby 2.[7](https://github.com/SketchUp/testup-2/actions/runs/7397795886/job/20125579628#step:4:8).8) [x86_64-darwin20]
  - rubocop-performance 1.15.2
```

**Comment by thomthom** [comment]
Turns out it was an issue with how Bundle installed gems into a `vendor/` folder directly in the project root.

Got a clue for another build where I switched to Ruby 3.2 and got a slightly different error message:

```
An error occurred while SketchupRequirements/GlobalConstants cop was inspecting /Users/runner/work/testup-2/testup-2/vendor/bundle/ruby/3.1.0/gems/unicode-display_width-2.5.0/lib/unicode/display_width/index.rb:11:6.
```

It was trying to process Ruby files in vendored gems installed by Bundler.
After excluding `vendor/` from RuboCop everything was working fine.

**Comment by trinistr** [comment]
If you found this issue from a web search:

This can happen if you specify `AllCops/Exclude`. It is an array option, so specifying it overrides [default configuration](https://github.com/rubocop/rubocop/blob/master/config/default.yml) which is supposed to prevent exactly this issue.

Possible solutions:
- [use inherit_mode](https://docs.rubocop.org/rubocop/configuration.html#merging-arrays-using-inherit_mode),
- add default values,
- disable RuboCop for the file using [directives](https://docs.rubocop.org/rubocop/configuration.html#disabling-cops-within-source-code).

**Comment by juharris** [comment]
Fix for me:
In `.rubocop.yml`:
```YAML
AllCops:
  Exclude:
    ...
    - "vendor/**/*"
```
